cloudmarker/events/gcpiamcorpruleevent.py

```diff
@@ -0,0 +1,82 @@
+"""IAM-Policy rule event.
+
+This module defines the :class:`GCPIAMCorpRuleEvent` class that identifies
+weak IAM-Policy related rules. This plugin works on the GCP IAM properties
+found in the ``com`` bucket of IAM-Policy rule records.
+"""
+
+
+import logging
+
+from cloudmarker import util
+
+_log = logging.getLogger(__name__)
+
+
+class GCPIAMCorpRuleEvent:
+    """IAM-Policy rule event plugin."""
+
+    def __init__(self):
+        """Create an instance of :class:`GCPIAMCorpRuleEvent` plugin."""
+
+    def eval(self, record):
+        """Evaluate IAM rules to check corporate login credentials.
+
+        Arguments:
+            record (dict): A IAM-Corp-Login-Policy rule record.
+
+        Yields:
+            dict: An event record representing a personal Gmail accounts.
+
+        """
+        # If 'com' bucket is missing, we have a malformed record. Log a
+        # warning and ignore it.
+        com = record.get('com')
+        if com is None:
+            _log.warning('IAM-Policy rule record is missing com key: %r',
+                         record)
+            return
+
+        # This plugin understands IAM-Policy rule records only, so ignore
+        # any other record types.
+        common_record_type = com.get('record_type')
+        if common_record_type != 'iam_corp_login_policy_rule':
+            return
+
+        members = record['raw']['members']
+        personal_account = None
+        for member in members:
+            if 'user' in member:
+                user = member.split('user:')
+                if user[1].endswith('gmail.com'):
```

Comment:
The user could be using any other mailing account as well, apart from `gmail`. Should we add checks for that too and make this generalised? I may be wrong here so please correct me.